common/src/chain/config.rs

```diff
@@ -117,6 +117,21 @@ pub fn create_mainnet() -> ChainConfig {
     }
 }
 
+pub fn create_mainnet_with_netupgrades(net_upgrades: NetUpgrades<UpgradeVersion>) -> ChainConfig {
```

Comment:
This is very dangerous. Mainnet parameters should not be parametrized in any way and should be 100% determined at compile-time. If this is for testing, please create a separate net for unit tests like we did today.

Reply:
Removed it.